crates/store/re_types/src/datatypes/pixel_format_ext.rs

```diff
@@ -24,24 +42,57 @@ impl PixelFormat {
     pub fn num_bytes(&self, [w, h]: [u32; 2]) -> usize {
         let num_pixels = w as usize * h as usize;
         match self {
-            Self::NV12 => 12 * num_pixels / 8,
-            Self::YUY2 => 16 * num_pixels / 8,
+            // 444 formats.
+            Self::Y_U_V24_FullRange | Self::Y_U_V24_LimitedRange => num_pixels * 4,
+
+            // 422 formats.
+            Self::Y_U_V16_FullRange | Self::Y_U_V16_LimitedRange | Self::NV12 => {
+                16 * num_pixels / 8
+            }
+
+            // 420 formats.
+            Self::Y_U_V12_FullRange | Self::Y_U_V12_LimitedRange | Self::YUY2 => {
+                12 * num_pixels / 8
+            }
+
+            // Monochrome formats.
+            Self::Y8_LimitedRange | Self::Y8_FullRange => num_pixels,
         }
     }
 
     /// The color model derived from this pixel format.
     #[inline]
     pub fn color_model(&self) -> ColorModel {
+        #[allow(clippy::match_same_arms)]
         match self {
-            Self::NV12 | Self::YUY2 => ColorModel::RGB,
+            Self::Y_U_V12_FullRange
+            | Self::Y_U_V16_FullRange
+            | Self::Y_U_V24_FullRange
+            | Self::Y_U_V12_LimitedRange
+            | Self::Y_U_V16_LimitedRange
+            | Self::Y_U_V24_LimitedRange
+            | Self::NV12
+            | Self::YUY2 => ColorModel::RGB,
+
+            // TODO(andreas): This shouldn't be ColorModel::RGB, but our YUV converter can't do anything else right now.
+            Self::Y8_LimitedRange | Self::Y8_FullRange => ColorModel::RGB,
```

Comment:
Can you elaborate on this?

You mean the GPU always outputs RGB, so we consider `Y` to be a very stupid way of compressing RGB?

What is `color_model` used for? Is there anything that will break due to this "lie"?

Reply:
(will add to comment)
GPU doesn't _have_ to always output RGB, but having it sometimes output R(8) _specifically_ for the YUV converter requires me to do more bookkeeping (needs a new renderpipeline and I expect other ripples) which I just don't want to do right now.
We do use this property mostly for things in the hover logic afaik.

> very stupid way of compressing RGB

hm well. no. That's just because our gpu conversion pipeline is stupid atm. But it still doesn't take up extra CPU memory and GPU textures are only held as long as they are in use (image cache for instance doesn't cache gpu transfer step!)